test/parser/LiteralOrIriTest.cpp

```diff
@@ -171,14 +170,25 @@ TEST(LiteralOrIri, LiteralOrIriWithLiteralWithQuotesAndLanguageTag) {
 TEST(LiteralOrIri, GetContent) {
   LiteralOrIri iri = LiteralOrIri::iriref("<https://example.org/books/book1>");
   LiteralOrIri literalWithLanguageTag =
-      LiteralOrIri::literalWithoutQuotesWithLanguageTag("Hello World", "@de");
-  LiteralOrIri literalWithDatatype =
-      LiteralOrIri::literalWithoutQuotesWithDatatype(
-          "ABC", Iri::iriref("<https://example.org>"));
+      LiteralOrIri::literalWithoutQuotes("Hello World", "@de");
+  LiteralOrIri literalWithDatatype = LiteralOrIri::literalWithoutQuotes(
+      "ABC", Iri::iriref("<https://example.org>"));
 
   EXPECT_THAT("https://example.org/books/book1",
               asStringViewUnsafe(iri.getContent()));
   EXPECT_THAT("Hello World",
               asStringViewUnsafe(literalWithLanguageTag.getContent()));
   EXPECT_THAT("ABC", asStringViewUnsafe(literalWithDatatype.getContent()));
 }
+
+TEST(LiteralOrIri, EnsureLiteralsAreEncoded) {
+  LiteralOrIri literal1 =
+      LiteralOrIri::literalWithQuotes(R"("This is to be \"\t encoded")");
+  EXPECT_THAT("This is to be \"\t encoded",
+              asStringViewUnsafe(literal1.getContent()));
+
+  LiteralOrIri literal2 =
+      LiteralOrIri::literalWithoutQuotes(R"(This is to be \"\t encoded)");
+  EXPECT_THAT("This is to be \"\t encoded",
+              asStringViewUnsafe(literal2.getContent()));
```

Comment:
Please don't use the Raw literals here, it is really unintuitive that way. If the original version has a lot of backslahses and the normalized version has less, then it is easier to understand.